docs/Using-SRIOV-and-Clear-Containers.md

```diff
@@ -0,0 +1,281 @@
+machines or containers. For Clear Containers, we enabled a Container Network 
+Model (CNM) plugin. Additionally, we made the necessary changes in the 
+runtime to detect virtual functions in a container's network namespace to be 
+able to use SR-IOV for network based devices.
+
+## Install the SR-IOV Docker\* plugin
+
+To create a network with VFs associated with it, that can be passed to
+Clear Containers, it is necessary to install a SR-IOV Docker plugin. The 
+created network is based on a physical function (PF) device, The network can
+create 'n' containers, where 'n' is the number of VFs associated with the 
+Physical Function (PF).
+
+To install the plugin, follow the [plugin installation instructions](https://github.com/clearcontainers/sriov).
+
+
+## Host setup for SR-IOV
+
+In order to setup your host for SR-IOV, the following has to be true:
+
+- The host system must support Intel VT-d. 
+- The host kernel needs to have Input-Output Memory Management Unit (IOMMU) 
+  and Virtual Function I/O (VFIO) support.
+- `CONFIG_VFIO_NOIOMMU` must to be disabled in the host kernel
+  configuration. You will need to rebuild your host system's kernel in
+  order to disable `CONFIG_VFIO_NOIOMMU` in the kernel configuration.
+- Optionally, you might need to add a PCI override for your Network Interface 
+  Controller (NIC). The immediately following section describes how to assess 
+  if you need to make NIC changes and how to make the necessary changes.
+
+### IOMMU Groups and PCIe Access Control Services
+
+Run the following command to see how the IOMMU groups are setup on your 
+host system. 
+```
+$ find /sys/kernel/iommu_groups/ -type l
+```
+
+The command's output details whether or not your NIC is setup 
+appropriately with respect to PCIe Access Control Services (ACS). If the PCI 
+bridge is within the same IOMMU group as your NIC, then this is an indication 
+that either your device doesn't support ACS or that it doesn't share this 
+capability appropriately by default.
+
+If the IOMMU groups are setup properly, you will have the PCI for each ACS enabled 
+NIC port in its own IOMMU group. If you do not see any output when running the above
+command, then you likely need to update your host's kernel configuration.
+
+For more details, see http://vfio.blogspot.com/2014/08/iommu-groups-inside-and-out.html
+
+### Update the host kernel
+
+
+Depending on your host kernel configuration, you might have to rebuild the 
+kernel. If the following conditions are true, you do not need to rebuild 
+your kernel. 
+
+- `CONFIG_VFIO_IOMMU_TYPE1`, `CONFIG_VFIO`, and `CONFIG_VFIO_PCI` are set in 
+the kernel configuration. Your kernel is built with VFIO support when 
+configurations are set. 
+- `CONFIG_VFIO_NOIOMMU` is disabled in the host kernel configuration.
+
+See the following steps one through three if you need to rebuild the kernel.
+
+The following steps update the SR-IOV host system. The steps are based on 
```

Comment:
Lines 69-71 suggested rewrite: The following steps, which are based on the Ubuntu 16.04 distribution, update the SR-IOV host system. If you use a different distribution, make appropriate adjustments to the commands: